analyzers/src/SonarAnalyzer.Common/Trackers/ArgumentTracker.cs

```diff
@@ -0,0 +1,86 @@
+﻿/*
+* SonarAnalyzer for .NET
+* Copyright (C) 2015-2023 SonarSource SA
+* mailto: contact AT sonarsource DOT com
+*
+* This program is free software; you can redistribute it and/or
+* modify it under the terms of the GNU Lesser General Public
+* License as published by the Free Software Foundation; either
+* version 3 of the License, or (at your option) any later version.
+*
+* This program is distributed in the hope that it will be useful,
+* but WITHOUT ANY WARRANTY; without even the implied warranty of
+* MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+* Lesser General Public License for more details.
+*
+* You should have received a copy of the GNU Lesser General Public License
+* along with this program; if not, write to the Free Software Foundation,
+* Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+*/
+
+namespace SonarAnalyzer.Helpers.Trackers;
+
+public abstract class ArgumentTracker<TSyntaxKind> : SyntaxTrackerBase<TSyntaxKind, ArgumentContext>
+    where TSyntaxKind : struct
+{
+    protected abstract RefKind? ArgumentRefKind(SyntaxNode argumentNode);
+    protected abstract IReadOnlyCollection<SyntaxNode> ArgumentList(SyntaxNode argumentNode);
+    protected abstract int? Position(SyntaxNode argumentNode);
+    protected abstract bool InvocationFitsMemberKind(SyntaxNode invokedExpression, MemberKind memberKind);
+    protected abstract bool InvokedMemberFits(SemanticModel model, SyntaxNode invokedExpression, MemberKind memberKind, Func<string, bool> invokedMemberNameConstraint);
```

Comment:
pedantic, for consistency:

- `InvokedMemberFits` -> `InvokeMemberMatches`
- `ParameterFits` -> `ParameterMatches`
- `InvocationFitsMemberKind` -> `InvocationMatchesMemberKind`